### PR TITLE
Config validation

### PR DESCRIPTION
**Pull request by ikbalkaya**
This PR adds some validators to required Ably configurations. 

Closes https://github.com/ably/kafka-connect-ably/issues/53

### CODE REVIEW

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java [hunk 1]
```diff
@@ -57,7 +55,49 @@ public void testConnector_connectorFailsWithNoTopicsGiven() {
...
+        assertThrows(ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
+    }
+
+    @Test
+    public void testConnector_connectorFailsWithNoClientIdGiven() {
+        final String channelName = "test-channel";
+
+        Map<String, String> settings = createSettings(channelName, "some_fake_key", null, TOPICS);
+        assertThrows(ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
+    }
```
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java:67]: I'd suggest refactoring the new tests with the "GIVEN WHEN THEN" approach and make sure that the important things for a given test are highlighted :wink: For example, here you're declaring a `channelName` at the start of the test, so I'm already thinking "okay, we'll be testing how this channel name affects the behaviour" but in this test the channel name is not important :sweat_smile: After the suggested changes this test could look like this for example:
```java
    @Test
    public void testConnector_connectorFailsWithNoClientIdGiven() {
        // given
        final String clientId = null;

        // when
        Map<String, String> settings = createSettings("test-channel", "some_fake_key", clientId, TOPICS);

        // then
        assertThrows(ConnectRestException.class, () -> connectCluster.configureConnector(CONNECTOR_NAME, settings));
    }
```

File: src/test/java/com/ably/kafka/connect/validators/ChannelNameValidatorTest.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
+package com.ably.kafka.connect.validators;
+
+import org.apache.kafka.common.config.ConfigException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+class ChannelNameValidatorTest {
```
Comment (KacperKluka) [inline src/test/java/com/ably/kafka/connect/validators/ChannelNameValidatorTest.java:8]: We have tests for `:` and `[` but there are some other invalid characters left from the regex pattern (namely whitespace and `,`), I think we should test them too as they feel as much important as the ones that are already tested :wink:

**Review by KacperKluka — CHANGES_REQUESTED** [review]
Functionally everything looks good :+1: I've left some suggestions regarding tests and naming :wink:

File: src/main/java/com/ably/kafka/connect/validators/AblyConfigValidators.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.ably.kafka.connect.validators;
+
+import org.apache.kafka.common.config.ConfigDef;
+import javax.annotation.Nonnull;
+
+/*
+Currently connector configurations take only a single validator per configuration. This class intends to combine and
+support multiple validators.
+* */
+public class AblyConfigValidators implements ConfigDef.Validator {
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/validators/AblyConfigValidators.java:10]: This is a good idea. I renamed it here 50ca79c1588395538a60d75d0a7fca0766b2214f

File: src/main/java/com/ably/kafka/connect/validators/AblyConfigValidators.java [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+   /**
+    * @param validators validators to use. Please provide them by their priority of importance.
+    */
+    public AblyConfigValidators(@Nonnull ConfigDef.Validator[] validators) {
+        this.validators = validators;
+    }
+
+    @Override
+    public void ensureValid(String name, Object value) {
+        if (validators.length > 0) {
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/validators/AblyConfigValidators.java]: You are absolutely right, I haven't been using regular arrays for a while 😄 6496203f15dc5ab6ed790f01ab3f2f9bcf591c22

File: src/main/java/com/ably/kafka/connect/ChannelSinkConnectorConfig.java [hunk 5]
```diff
@@ -294,18 +297,23 @@ public static ConfigDef createConfig() {
...
                     .documentation(CLIENT_KEY_DOC)
                     .importance(Importance.HIGH)
+                    .validator(new ConfigDef.NonNullValidator())
                     .build()
             )
             .define(
                 ConfigKeyBuilder.of(CLIENT_ID, Type.STRING)
                     .documentation(CLIENT_ID_DOC)
                     .importance(Importance.HIGH)
+                    .validator(new AblyConfigValidators(new ConfigDef.Validator[]{new ConfigDef.NonNullValidator(), new ConfigDef.NonEmptyString()}))
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/ChannelSinkConnectorConfig.java]: 497b51a35c759f740622162ccec933b7aea237d7

File: src/main/java/com/ably/kafka/connect/ChannelSinkConnectorConfig.java [hunk 6]
```diff
@@ -294,18 +297,23 @@ public static ConfigDef createConfig() {
                 ConfigKeyBuilder.of(CHANNEL_CONFIG, Type.STRING)
                     .documentation(CHANNEL_CONFIG_DOC)
                     .importance(Importance.HIGH)
+                    .validator(new AblyConfigValidators(new ConfigDef.Validator[]{new ConfigDef.NonNullValidator(),
+                        new ConfigDef.NonEmptyString(),
+                        new ChannelNameValidator()}))
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/ChannelSinkConnectorConfig.java]: 497b51a35c759f740622162ccec933b7aea237d7

File: src/main/java/com/ably/kafka/connect/validators/ChannelNameValidator.java [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+import java.util.regex.Pattern;
+
+/**
+ * This validator should be used to check the format of the channel name.
+ * Other priority validators must be used first
+ * */
+public class ChannelNameValidator implements ConfigDef.Validator {
+    @Override
+    public void ensureValid(String name, Object value) {
+        final String patternString = "^([^:,\\s\\[].*)$";
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/main/java/com/ably/kafka/connect/validators/ChannelNameValidator.java]: This is also a good idea. Here it is e5402f80e087a44c9144784098a046de9388b0e8

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java [hunk 8]
```diff
@@ -12,10 +13,7 @@
 import java.util.HashMap;
 import java.util.Map;
 
-import static org.apache.kafka.connect.runtime.ConnectorConfig.CONNECTOR_CLASS_CONFIG;
-import static org.apache.kafka.connect.runtime.ConnectorConfig.KEY_CONVERTER_CLASS_CONFIG;
-import static org.apache.kafka.connect.runtime.ConnectorConfig.TASKS_MAX_CONFIG;
-import static org.apache.kafka.connect.runtime.ConnectorConfig.VALUE_CONVERTER_CLASS_CONFIG;
+import static org.apache.kafka.connect.runtime.ConnectorConfig.*;
```
[the remark this answers is not in the document]
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java]: 7b2175f21596cb9d956e1690a4d0e56440dfcce5

File: src/test/java/com/ably/kafka/connect/validators/ChannelNameValidatorTest.java (hunk 2 above)
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/validators/ChannelNameValidatorTest.java:8]: b511308bd68a1556a594d80e032e6b3c121e4268

File: src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java (hunk 1 above)
Comment (ikbalkaya) [reply src/test/java/com/ably/kafka/connect/integration/ChannelSinkConnectorTest.java:71]: @KacperKluka A bit busy right now, but I shall put those onto organised states later in another PR

**Review by KacperKluka — APPROVED** [review]
LGTM :+1:

**Pull request merged**